audio/src/convert.rs

```diff
@@ -12,45 +14,72 @@ impl i24 {
     }
 }
 
-// Losslessly represent [-1.0, 1.0] to [$type::MIN, $type::MAX] while maintaining DC linearity.
+pub struct Requantizer {
+    ditherer: Box<dyn Ditherer>,
+    noise_shaper: Box<dyn NoiseShaper>,
+}
+
+impl Requantizer {
+    pub fn new(ditherer: Box<dyn Ditherer>, noise_shaper: Box<dyn NoiseShaper>) -> Self {
+        info!(
+            "Requantizing with ditherer: {} and noise shaper: {}",
+            ditherer, noise_shaper
+        );
+        Self {
+            ditherer,
+            noise_shaper,
+        }
+    }
+
+    pub fn shaped_dither(&mut self, sample: f32) -> f32 {
+        let noise = self.ditherer.noise(sample);
+        self.noise_shaper.shape(sample, noise)
+    }
```

Comment:
I think the `Ditherer` and `NoiseShaper` traits should have default functions that take buffers of samples and modify them, and shaped_dither should take buffers itself and call these functions instead to reduce the number of dynamically dispatched function calls. (On the other hand, there seems not to be a performance problem, so it's not that important)